src/main/java/edu/illinois/cs/dt/tools/fixer/CleanerFixerPlugin.java

```diff
@@ -0,0 +1,211 @@
+
+        if (!cleanerMethodOpt.isPresent()) {
+            TestPluginPlugin.error("Could not find method " + cleanerTestName);
+            TestPluginPlugin.error("Tried looking in: " + testFiles);
+            return;
+        }
+
+        if (!victimMethodOpt.isPresent()) {
+            TestPluginPlugin.error("Could not find method " + victimTestName);
+            TestPluginPlugin.error("Tried looking in: " + testFiles);
+            return;
+        }
+
+        applyFix(polluterTestName, cleanerMethodOpt.get(), victimMethodOpt.get());
+    }
+
+    private void backup(final JavaFile javaFile) throws IOException {
+        final Path path = CleanerPathManager.backupPath(javaFile.path());
+        Files.copy(javaFile.path(), path, StandardCopyOption.REPLACE_EXISTING);
+    }
+
+    // TODO: Extract this logic out to a more generalized fixer that is separate, so we can reuse it
+    // TODO: for cleaners, santa clauses, etc.
+    private void applyFix(final String polluterName,
+                          final JavaMethod cleanerMethod,
+                          final JavaMethod victimMethod) throws Exception {
+        backup(victimMethod.javaFile());
+
+        // Compile the test without our fix
+//        tryRecompile(victimMethod.javaFile());
+
+        runMvnInstall();
+
+        // Check if we pass in isolation before fix
+        // TODO: We should get rid of this check eventually, because it won't work for the case of \santaClause
+        TestPluginPlugin.info("Running victim test in isolation without code from cleaner.");
+        if (!testOrderPasses(Collections.singletonList(victimMethod.methodName()))) {
+            TestPluginPlugin.error("Test fails in isolation");
+            return;
+        }
+
+        // Do our fix
+        TestPluginPlugin.info("Applying code from cleaner and recompiling.");
+        final NodeList<Statement> cleanerStmts = cleanerMethod.body().getStatements();
+
+        // Note: this modifies victimMethod, so when we eventually make it delta-debug, we will want
+        // to make a copy of the victimMethod somewhere
+        victimMethod.prepend(cleanerStmts);
+        victimMethod.javaFile().writeAndReloadCompilationUnit();
+
+        // Check if we pass in isolation after fix
+//        tryRecompile(victimMethod.javaFile());
+        runMvnInstall();
+
+        // TODO: Output to result files rather than stdout
+        TestPluginPlugin.info("Running victim test in isolation with code from cleaner.");
+        boolean passInIsolationAfterFix = testOrderPasses(ListUtil.fromArray(polluterName, victimMethod.methodName()));
+        if (!passInIsolationAfterFix) {
+            TestPluginPlugin.error("Fix was unsuccessful. Test still fails with polluter.");
+        } else {
+            TestPluginPlugin.info("Fix was successful! Fixed file:\n" + victimMethod.javaFile().path());
+        }
+
+        // TODO: What is this for?
```

Comment:
I don't think we need this anymore, when trying to delta-debug I did not need to uncomment this code.

Reply:
This block of code could be useful when dealing with a case where we add the necessary setup to a victim test but our "fix" would actually now cause another test to fail. We would need some additional logic to deal with this case (i.e., a fix reveals another dependency) than just this block of code, but it may have its uses later.